nixos/modules/misc/machine-info.nix

```diff
@@ -0,0 +1,83 @@
+{ config, lib, ... }:
+let
+  cfg = config.environment.machineInfo;
+
+  needsEscaping = s: builtins.match "[a-zA-Z0-9]+" s == null;
+  escapeIfNecessary = s: if needsEscaping s then ''"${lib.escape [ "\$" "\"" "\\" "\`" ] s}"'' else s;
+  attrsToText = attrs:
+    lib.concatLines (lib.mapAttrsToList
+      (name: value: "${name}=${escapeIfNecessary value}")
+      (lib.filterAttrs (_: value: value != null) attrs));
+
+  text = attrsToText {
+    PRETTY_HOSTNAME = cfg.prettyHostname;
+    ICON_NAME = cfg.iconName;
+    CHASSIS = cfg.chassis;
+    DEPLOYMENT = cfg.deployment;
+    LOCATION = cfg.location;
+    HARDWARE_VENDOR = cfg.hardwareVendor;
+    HARDWARE_MODEL = cfg.hardwareModel;
+  };
+
+  mkVarOption = description: lib.mkOption {
+    type = lib.types.nullOr lib.types.singleLineStr;
+    default = null;
+    description = lib.mdDoc description;
+  };
+in
+{
+  options.environment.machineInfo = {
```

Comment:
`systemd.etc.machineInfo` seems appropriate since it goes to `etc`, and there are also files like [os-release] that we may want to add in the future.

[os-release]: https://www.freedesktop.org/software/systemd/man/os-release.html

Reply:
I think `systemd.machineInfo` is sufficient. The `etc` seems redundant here.
Btw. `os-release` is already auto generated.